src/mpid/ch4/src/ch4r_recvq.h

```diff
@@ -28,9 +28,39 @@ int MPIDIG_recvq_init(void);
 MPL_STATIC_INLINE_PREFIX int MPIDIG_match_posted(int rank, int tag,
                                                  MPIR_Context_id_t context_id, MPIR_Request * req)
 {
-    return (rank == MPIDIG_REQUEST(req, rank) || MPIDIG_REQUEST(req, rank) == MPI_ANY_SOURCE) &&
+    /* FIXME: anysource_partner_request need use atomics */
+    int match;
+    match = (rank == MPIDIG_REQUEST(req, rank) || MPIDIG_REQUEST(req, rank) == MPI_ANY_SOURCE) &&
         (tag == MPIR_TAG_MASK_ERROR_BITS(MPIDIG_REQUEST(req, tag)) ||
          MPIDIG_REQUEST(req, tag) == MPI_ANY_TAG) && context_id == MPIDIG_REQUEST(req, context_id);
+    if (!match) {
+        return 0;
+    } else if (req->kind == MPIR_REQUEST_KIND__ANYSRC_RECV) {
+        MPIR_Request *req2;
+        req2 = MPIR_REQUEST_ANYSRC_PARTNER(req);
+        if (req2 == NULL) {
+            /* still in the process of creating netmod recv request */
+            return 0;
```

Comment:
If you want to have different return values mean different things, can you make them an `enum` so it's easier to understand?

Reply:
I am not saying I am happy with this code. The problem is that the code is still not easy to understand using names. :)

Reply:
Maybe not, but we purposefully went through all of the code and change the use of integer constants recently to move them all to enums to improve readability, even in the case of simple things like 0/1 for true/false.